src/nix/app.cc

```diff
@@ -20,20 +20,24 @@ StringPairs resolveRewrites(
     const std::vector<BuiltPathWithResult> & dependencies)
 {
     StringPairs res;
-    if (experimentalFeatureSettings.isEnabled(Xp::CaDerivations)) {
-        for (auto & dep : dependencies) {
-            if (auto drvDep = std::get_if<BuiltPathBuilt>(&dep.path)) {
-                for (auto & [ outputName, outputPath ] : drvDep->outputs) {
-                    res.emplace(
-                        DownstreamPlaceholder::fromSingleDerivedPathBuilt(
-                            SingleDerivedPath::Built {
-                                .drvPath = make_ref<SingleDerivedPath>(drvDep->drvPath->discardOutputPath()),
-                                .output = outputName,
-                            }).render(),
-                        store.printStorePath(outputPath)
-                    );
-                }
-            }
+    if (!experimentalFeatureSettings.isEnabled(Xp::CaDerivations)) {
+        return res;
+    }
+    for (auto &dep: dependencies) {
+        auto drvDep = std::get_if<BuiltPathBuilt>(&dep.path);
+        if (!drvDep) {
+            continue;
+        }
+
+        for (auto & [ outputName, outputPath ] : drvDep->outputs) {
```

Comment:
it seems like this could even be const.

Reply:
Yep, I'll commit that too a little bit later.